### PR TITLE
[integration] Remove data dir

### LINKED ISSUE #495: [integration] Remove `data` dir and retrieve at test time
The most up-to-date version of our project data (examples, models, etc.) is found in our Google Drive folder: https://drive.google.com/drive/folders/1RnNVTifUESFXXkNh3tc8J6kot0xink07?usp=drive_link

Following what we did with models in #391 , we could retrieve gromet examples and individual model files from the bucket by regularly syncing it with our Google Drive data.

This will help ensure examples remain up-to-date which is especially important given the gromet data model is under active development.  We also avoid bloating the repo with large data diffs.

### PR DESCRIPTION
**Pull request by myedibleenso**
## Summary of changes

Removes the data dir in favor of a periodic sync of gromet and model data stored on in Google Drive folder: https://drive.google.com/drive/folders/1RnNVTifUESFXXkNh3tc8J6kot0xink07?usp=drive_link

... to our s3 bucket.

Resolves #495

### CODE REVIEW

**Comment by myedibleenso** [comment]
I think that just leaves the Rust tests.  Addressing those will probably involve an additional dependency (something like the Rust equivalent of Python's `requests`).

**Comment by myedibleenso** [comment]
@Free-Quarks , note that https://github.com/ml4ai/skema/tree/5e5bba8631054d9ac2ab99f2fdb4f0a4f5561d8d/data/mml2pn_inputs has been moved into `skema/skema-rs/mathml/tests`

**Review by vincentraymond-ua — APPROVED** [review]
@myedibleenso - Looks good! I've also fixed and re-enabled the comment_extractor tests. Turns out we weren't building any tree-sitter parsers in the workflow, so I've added an additional step for that.

File: .github/workflows/tests-and-docs.yml [hunk 1]
```diff
@@ -57,7 +57,11 @@ jobs:
         # retrieve latest model for img2mml component
         curl -L https://artifacts.askem.lum.ai/skema/img2mml/models/cnn_xfmer_arxiv_im2mml_with_fonts_boldface_best.pt > skema/img2mml/trained_models/cnn_xfmer_arxiv_im2mml_with_fonts_boldface_best.pt
         pip install ".[all]"
-
+    # Install tree-sitter parser (for Python component unit tests)
+    - name: Install tree-sitter parsers
+      working-directory: .
+      run: python skema/program_analysis/tree_sitter_parsers/build_parsers.py --c --cpp --fortran --python --matlab --r
```
Comment (myedibleenso) [inline .github/workflows/tests-and-docs.yml:63]: Thanks, @vincentraymond-ua !  Can you also update the dev environment documentation?  https://ml4ai.github.io/skema/dev/env/#configuring-your-dev-environment

Any idea why the matlab and fortran tests were passing?

**Review by Free-Quarks — APPROVED** [review]
Looks good to me too!

**Comment by myedibleenso** [comment]
@enoriega , can you please take a look at https://github.com/ml4ai/skema/actions/runs/6295232322/job/17088202803?pr=494#step:13:71 ?

**Comment by enoriega** [comment]
One machine that serves the request is down, we're looking into it @myedibleenso

**Comment by enoriega** [comment]
@myedibleenso This should be fixed now!

**Pull request merged**